### ISSUE TITLE
Can

### ISSUE THREAD

**Issue by palashman** [jacquesh/foo_openlyrics#77]

**Comment by palashman** [comment]
Hello!

If you change track title tag, then plugin can't read present *.lrc file.

For example present title is "_C'est La Vie_" and then you rename it to, say, "_C'est La Vie (that's life)_", id est you add little suffix in brackets. As for now to make plugin works in mentioned case, I have to manually rename lrc-file from "_Metallica - C'est La Vie.lrc_" to "_Metallica - C'est La Vie (that's life).lrc_"

It'd be helpful and cosy if plugin could understand present *.lrc even if you made insignificant title tag editing in brackets.

Thank you in advance!

**Comment by johbiii** [comment]
Perhaps we can create a hash of the file on scanning for lyrics so that we have a tag to use for the database?

I was using md5 tags for my songs but not all containers support it.

**Comment by jacquesh** [comment]
Are you using foobar2000 itself to do the tag editing?

Off the top of my head I don't know if the SDK tells you when tags get changed but if they do I could have it ask if you want to also rename the tag file at the same time.

**Comment by johbiii** [comment]
> Are you using foobar2000 itself to do the tag editing?
> 
> Off the top of my head I don't know if the SDK tells you when tags get changed but if they do I could have it ask if you want to also rename the tag file at the same time.

The better way in my opinion would be to do a proper database.

Two methods for tracking would be by path and by tags. If you track using tags and have different editions of the same song it could throw off the timing for synced lyrics.

The Playback Statistics compnent (by the actual foobar dev) uses a database and automatically transfers data when changing tags. 

> When editing tags, affected playback statistics records are transferred accordingly.
[source](https://wiki.hydrogenaudio.org/index.php?title=Foobar2000:Components/Playback_Statistics_v3.x_(foo_playcount))

Unfortunately it is not open sourced, I'm not sure if there are any open SQLite components on GitHub to use as an example.

**Comment by palashman** [comment]
What if foo_openlyrics would have option to ignore tag's text in brackets: (), [], {} ?
Although some times brackets could contain precise information, like "Magical (Radio version)", "Magical (Remix)", "Magical (feat John Bo)". And for the last cases brackets must be considered. But if brackets contain just user lemma, they must be ignored. 
The point is how to differentiate the 1st case from the 2nd?

**Comment by jacquesh** [comment]
> The better way in my opinion would be to do a proper database.

Yeah maybe. I *think* this is what LSP3 called "Association search". fb2k has a way of letting you store some info about each track in your library (I'm using this for other things already). I was kinda hoping to get away with not doing that though because it seems like a whole bunch of effort for minimal gain.

If I can detect tag changes and it's simpler then I still think it'd be better to do that and consider adding a database-type lookup separately/later

**Comment by palashman** [comment]
Also such additions in title (in brackets) aggravate internet searching appropriate lyrics. Often you have to make manual searching.
